### DISCUSSION TITLE
Fault on search node after upgraded to 2.3.50

### DISCUSSION THREAD

**Discussion by byronsims** [Security-Onion-Solutions/securityonion#4250] [Unsupported Versions]
I upgraded to 2.3.50 and now I have one search node that is showing "Fault" in the grid view.  Running so-status on the search node shows everything as OK.

Checking Docker status

    Docker ----------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    

Checking container statuses

    so-curator ------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-elasticsearch ------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-filebeat ------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-logstash ------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-nginx --------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-sensoroni ----------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-telegraf ------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-wazuh --------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]  

All IDs are lowercase.  I saw that as a possible issue in another thread.

**Comment by trocade** [comment]
I had a similar issue yesterday and it was due to a fauly salt-minion installation package.
This is running Ubuntu 18.04 LTS

My steps to fix this:
1. systemctl stop salt-minion
2. ps -ef | grep -i salt (which showed me a few salt process but mainly a wget hanging on gpg keys)
3. kill above processes
4. apt upgrade -> got me a update for salt packages
5. systemctl start salt-minion

Your issue could be like mine if all is green but a `docker ps` will show that you are actually on a older version

**Comment by byronsims** [comment]
I'm running on centos.  yum update does not show me any salt packages needing to be updated.

docker ps shows:

CONTAINER ID   IMAGE                                                             COMMAND                  CREATED       STATUS        PORTS                                                                                                      NAMES
8b030bbca766   master:5000/security-onion-solutions/so-filebeat:2.3.50        "/usr/local/bin/dock…"   2 weeks ago   Up 25 hours   0.0.0.0:514->514/tcp, 0.0.0.0:5066->5066/tcp, 0.0.0.0:514->514/udp                                         so-filebeat
8944dd96e77e   master:5000/security-onion-solutions/so-curator:2.3.50         "/entrypoint.sh"         2 weeks ago   Up 25 hours                                                                                                              so-curator
7e1ecfc84aaf   master:5000/security-onion-solutions/so-logstash:2.3.50        "/usr/local/bin/dock…"   2 weeks ago   Up 25 hours   0.0.0.0:5044->5044/tcp, 0.0.0.0:5644->5644/tcp, 0.0.0.0:6050-6053->6050-6053/tcp, 0.0.0.0:9600->9600/tcp   so-logstash
d7300c607a45   master:5000/security-onion-solutions/so-elasticsearch:2.3.50   "/bin/tini -- /usr/l…"   2 weeks ago   Up 25 hours   0.0.0.0:9200->9200/tcp, 0.0.0.0:9300->9300/tcp                                                             so-elasticsearch
7ed4d37105d8   master:5000/security-onion-solutions/so-wazuh:2.3.50           "/entrypoint.sh"         2 weeks ago   Up 25 hours   0.0.0.0:1514-1515->1514-1515/tcp, 0.0.0.0:55000->55000/tcp, 0.0.0.0:1514->1514/udp                         so-wazuh
c771484cf391   master:5000/security-onion-solutions/so-telegraf:2.3.50        "/entrypoint.sh tele…"   2 weeks ago   Up 25 hours                                                                                                              so-telegraf
2d6370b176e0   master:5000/security-onion-solutions/so-nginx:2.3.50           "/docker-entrypoint.…"   2 weeks ago   Up 25 hours   0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp, 0.0.0.0:8090->8090/tcp                                           so-nginx
b64e48aa3902   master:5000/security-onion-solutions/so-soc:2.3.50             "/opt/sensoroni/sens…"   2 weeks ago   Up 25 hours                                                                                                              so-sensoroni

**Comment by jertel** [comment]
A screenshot showing the expanded node in the SOC grid page will help troubleshoot this.

**Reply by jertel** [reply]
Run this on your manager and let us know what you get back:
```
sudo docker exec -it so-influxdb influx -ssl -unsafeSsl -database telegraf -execute "select last(*) from sostatus group by host;"
```

**Reply by byronsims** [reply]
name: sostatus
tags: host=seccore
time                last_status
----                -----------
1623206220000000000 0

name: sostatus
tags: host=secinet
time                last_status
----                -----------
1623206160000000000 0

name: sostatus
tags: host=secmaster
time                last_status
----                -----------
1623206220000000000 0

name: sostatus
tags: host=secstorep1
time                last_status
----                -----------
1623191640000000000 0

name: sostatus
tags: host=secstorep2
time                last_status
----                -----------
1623206100000000000 0

name: sostatus
tags: host=secstorev1
time                last_status
----                -----------
1623205860000000000 0

name: sostatus
tags: host=secstorev2
time                last_status
----                -----------
1623206040000000000 0

**Reply by byronsims** [reply]
[output.txt](https://github.com/Security-Onion-Solutions/securityonion/files/6620731/output.txt)
I'll attach a text file with the output.  The copy/paste did some weird formatting.

**Reply by jertel** [reply]
Looks good. Check your search node's `/opt/so/conf/sensoroni/sensoroni.json` file for extra whitespace around the nodeId value. Ex:
```
    "agent": {
      "nodeId": "standalone",
      "role": "so-node",
```

**Reply by byronsims** [reply]
It looks correct to me, unless I'm missing something.
[sensoroni.txt](https://github.com/Security-Onion-Solutions/securityonion/files/6627128/sensoroni.txt)

**Reply by jertel** [reply]
Looks fine. Enable SOC debug, as shown in [this discussion](https://github.com/Security-Onion-Solutions/securityonion/discussions/2546#discussioncomment-270420). Then post back the log portion of the log starting with:

```
 DEBUG[0131] Got an influxdb result    result.Record=_start:2021-06-14 11:30:
```

**Reply by byronsims** [reply]
I added the "logLevel": "debug" to the top of the file right about the "logFilename", but whenever I run the so-soc-restart command, it takes it out.  The log does not seem to have any "debug" information in it afterward.

[so-soc-restart.txt](https://github.com/Security-Onion-Solutions/securityonion/files/6649040/so-soc-restart.txt)

**Reply by jertel** [reply]
Yes, salt manages the configuration file so applying the salt state by running something like `so-soc-start` will undo your modifications. If you instead make the json edit and then restart soc via `docker restart so-soc` you should see the debug logging for a few minutes. Salt automatically applies all states on a 15 minute interval so there's a small chance that as soon as you restart soc a high state could kick in and undo the config and restart soc again. However, doing it this way ensures no one accidentally leaves debug enabled permanently, which would eat through a lot more disk space than usual.

**Reply by byronsims** [reply]
IP for the node in question in the logs is 10.10.0.46.

[sensoroni-server2.log](https://github.com/Security-Onion-Solutions/securityonion/files/6650294/sensoroni-server2.log)

**Reply by jertel** [reply]
It looks like the secstorep1 node is not updating its status in a timely manner. SOC will only accept _recent_ status updates from all nodes, where by default, _recent_ means within the past 20 minutes. I just looked again at your output.txt file you attached last week from the influxdb query. That output shows that 6 nodes had provided an update around Wednesday, June 9, 2021 2:35:00 AM GMT, but the secstorep1 node's update is from Tuesday, June 8, 2021 10:34:00 PM. I suggest checking the `/opt/so/log/telegraf/telegraf.log` on secstorep1 and see if it's having trouble sending updates to influxdb on your manager node.

**Reply by byronsims** [reply]
[telegrafLog.txt](https://github.com/Security-Onion-Solutions/securityonion/files/6651104/telegrafLog.txt)

I do see some issues it looks like.  The attached are the only entries for today's log.

**Reply by jertel** [reply]
It looks like you have a network routing or firewall problem between secstorep1 and the manager.

**Reply by byronsims** [reply]
My storage node can connect via wget to https://secmaster:8086.

user@secstorep1 ~]$  wget --no-check-certificate  https://secmaster:8086
--2021-06-15 12:08:44--  https://secmaster:8086/
Resolving secmaster (secmaster)... 10.10.0.42
Connecting to secmaster (secmaster)|10.10.0.42|:8086... connected.
WARNING: cannot verify secmaster's certificate, issued by ‘/C=US/CN=secmaster/L=Salt Lake City/ST=Utah’:
  Unable to locally verify the issuer's authority.
HTTP request sent, awaiting response... 404 Not Found
2021-06-15 12:08:44 ERROR 404: Not Found.

**Reply by byronsims** [reply]
Is there anything else I can try?

**Reply by jertel** [reply]
You can use a packet sniffing tool like tcpdump or Wireshark, or perhaps your firewall has one built in, to help determine why the secstorep1 node can't consistently connect back to port 8086 on the manager.
